### ISSUE TITLE
Error InvalidConfigException Unable to locate message source for category 'typedlinkfield'.

### ISSUE THREAD

**Issue by Jones-S** [mmikkel/CpFieldInspect-Craft#14]
Hi there

I updated to the new Craft to use its core graphql service. As a consequence I had to update a few plugins. Like [Linkfield](https://github.com/sebastian-lenz/craft-linkfield) plugin.
After doing that I got an error, which I reported wrongly back to the maintainer of the linkfield repo. He told me that this was rather an issue with your plugin.

https://github.com/sebastian-lenz/craft-linkfield/issues/79
For the sake of completeness, the full stack trace:
```
yii\base\InvalidConfigException: Unable to locate message source for category 'typedlinkfield'. in /Users/USER/Documents/github/my-repo/vendor/yiisoft/yii2/i18n/I18N.php:201
Stack trace:
#0 /Users/USER/Documents/github/my-repo/vendor/yiisoft/yii2/i18n/I18N.php(89): yii\i18n\I18N->getMessageSource('typedlinkfield')
#1 /Users/USER/Documents/github/my-repo/vendor/craftcms/cms/src/i18n/I18N.php(327): yii\i18n\I18N->translate('typedlinkfield', 'Input fields', Array, 'en-US')
#2 /Users/USER/Documents/github/my-repo/vendor/yiisoft/yii2/BaseYii.php(526): craft\i18n\I18N->translate('typedlinkfield', 'Input fields', Array, 'en-US')
#3 /Users/USER/Documents/github/my-repo/vendor/sebastianlenz/linkfield/src/models/LinkType.php(101): yii\BaseYii::t('typedlinkfield', 'Input fields')
#4 /Users/USER/Documents/github/my-repo/vendor/sebastianlenz/linkfield/src/fields/LinkField.php(97): lenz\linkfield\models\LinkType->getTranslatedDisplayGroup()
#5 [internal function]: lenz\linkfield\fields\LinkField->lenz\linkfield\fields\{closure}(Object(lenz\linkfield\models\input\InputLinkType), Object(lenz\linkfield\models\input\InputLinkType))
#6 /Users/USER/Documents/github/my-repo/vendor/sebastianlenz/linkfield/src/fields/LinkField.php(100): uasort(Array, Object(Closure))
#7 /Users/USER/Documents/github/my-repo/vendor/sebastianlenz/linkfield/src/fields/LinkField.php(226): lenz\linkfield\fields\LinkField->getAvailableLinkTypes()
#8 /Users/USER/Documents/github/my-repo/vendor/yiisoft/yii2/base/Component.php(180): lenz\linkfield\fields\LinkField->setTypeSettings(Array)
#9 /Users/USER/Documents/github/my-repo/vendor/yiisoft/yii2/BaseYii.php(546): yii\base\Component->__set('typeSettings', Array)
#10 /Users/USER/Documents/github/my-repo/vendor/yiisoft/yii2/base/BaseObject.php(107): yii\BaseYii::configure(Object(lenz\linkfield\fields\LinkField), Array)
#11 /Users/USER/Documents/github/my-repo/vendor/craftcms/cms/src/helpers/Component.php(113): yii\base\BaseObject->__construct(Array)
#12 /Users/USER/Documents/github/my-repo/vendor/craftcms/cms/src/services/Fields.php(562): craft\helpers\Component::createComponent(Array, 'craft\\base\\Fiel...')
#13 /Users/USER/Documents/github/my-repo/vendor/craftcms/cms/src/services/Fields.php(587): craft\services\Fields->createField(Array)
#14 /Users/USER/Documents/github/my-repo/vendor/craftcms/cms/src/elements/db/ElementQuery.php(1941): craft\services\Fields->getAllFields()
#15 /Users/USER/Documents/github/my-repo/vendor/craftcms/cms/src/elements/db/ElementQuery.php(1364): craft\elements\db\ElementQuery->customFields()
#16 /Users/USER/Documents/github/my-repo/vendor/yiisoft/yii2/db/QueryBuilder.php(227): craft\elements\db\ElementQuery->prepare(Object(craft\db\mysql\QueryBuilder))
#17 /Users/USER/Documents/github/my-repo/vendor/yiisoft/yii2/db/Query.php(146): yii\db\QueryBuilder->build(Object(craft\elements\db\UserQuery))
#18 /Users/USER/Documents/github/my-repo/vendor/yiisoft/yii2/db/Query.php(274): yii\db\Query->createCommand(Object(craft\db\Connection))
#19 /Users/USER/Documents/github/my-repo/vendor/craftcms/cms/src/db/Query.php(177): yii\db\Query->one(NULL)
#20 /Users/USER/Documents/github/my-repo/vendor/craftcms/cms/src/elements/db/ElementQuery.php(1509): craft\db\Query->one(NULL)
#21 /Users/USER/Documents/github/my-repo/vendor/craftcms/cms/src/elements/User.php(383): craft\elements\db\ElementQuery->one()
#22 /Users/USER/Documents/github/my-repo/vendor/yiisoft/yii2/web/User.php(690): craft\elements\User::findIdentity('1')
#23 /Users/USER/Documents/github/my-repo/vendor/craftcms/cms/src/web/User.php(506): yii\web\User->renewAuthStatus()
#24 /Users/USER/Documents/github/my-repo/vendor/yiisoft/yii2/web/User.php(192): craft\web\User->renewAuthStatus()
#25 /Users/USER/Documents/github/my-repo/vendor/craftcms/cms/src/web/User.php(246): yii\web\User->getIdentity()
#26 /Users/USER/Documents/github/my-repo/vendor/mmikkel/cp-field-inspect/src/CpFieldInspect.php(80): craft\web\User->getIsAdmin()
#27 /Users/USER/Documents/github/my-repo/vendor/yiisoft/yii2/base/BaseObject.php(109): mmikkel\cpfieldinspect\CpFieldInspect->init()
#28 /Users/USER/Documents/github/my-repo/vendor/yiisoft/yii2/base/Module.php(158): yii\base\BaseObject->__construct(Array)
#29 /Users/USER/Documents/github/my-repo/vendor/craftcms/cms/src/base/Plugin.php(130): yii\base\Module->__construct('cp-field-inspec...', Object(craft\web\Application), Array)
#30 [internal function]: craft\base\Plugin->__construct('cp-field-inspec...', Object(craft\web\Application), Array)
#31 /Users/USER/Documents/github/my-repo/vendor/yiisoft/yii2/di/Container.php(384): ReflectionClass->newInstanceArgs(Array)
#32 /Users/USER/Documents/github/my-repo/vendor/yiisoft/yii2/di/Container.php(156): yii\di\Container->build('mmikkel\\cpfield...', Array, Array)
#33 /Users/USER/Documents/github/my-repo/vendor/yiisoft/yii2/BaseYii.php(349): yii\di\Container->get('mmikkel\\cpfield...', Array, Array)
#34 /Users/USER/Documents/github/my-repo/vendor/craftcms/cms/src/services/Plugins.php(897): yii\BaseYii::createObject(Array, Array)
#35 /Users/USER/Documents/github/my-repo/vendor/craftcms/cms/src/services/Plugins.php(230): craft\services\Plugins->createPlugin('cp-field-inspec...', Array)
#36 /Users/USER/Documents/github/my-repo/vendor/craftcms/cms/src/base/ApplicationTrait.php(1361): craft\services\Plugins->loadPlugins()
#37 /Users/USER/Documents/github/my-repo/vendor/craftcms/cms/src/web/Application.php(112): craft\web\Application->_postInit()
#38 /Users/USER/Documents/github/my-repo/vendor/yiisoft/yii2/base/BaseObject.php(109): craft\web\Application->init()
#39 /Users/USER/Documents/github/my-repo/vendor/yiisoft/yii2/base/Application.php(206): yii\base\BaseObject->__construct(Array)
#40 /Users/USER/Documents/github/my-repo/vendor/craftcms/cms/src/web/Application.php(100): yii\base\Application->__construct(Array)
#41 [internal function]: craft\web\Application->__construct(Array)
#42 /Users/USER/Documents/github/my-repo/vendor/yiisoft/yii2/di/Container.php(384): ReflectionClass->newInstanceArgs(Array)
#43 /Users/USER/Documents/github/my-repo/vendor/yiisoft/yii2/di/Container.php(156): yii\di\Container->build('craft\\web\\Appli...', Array, Array)
#44 /Users/USER/Documents/github/my-repo/vendor/yiisoft/yii2/BaseYii.php(349): yii\di\Container->get('craft\\web\\Appli...', Array, Array)
#45 /Users/USER/Documents/github/my-repo/vendor/craftcms/cms/bootstrap/bootstrap.php(244): yii\BaseYii::createObject(Array)
#46 /Users/USER/Documents/github/my-repo/vendor/craftcms/cms/bootstrap/web.php(52): require('/Users/USER/Do...')
#47 /Users/USER/Documents/github/my-repo/web/index.php(21): require('/Users/USER/Do...')
#48 {main}
```

Could you have a look at this? I find your plugin very helpful and I would like to keep using it.

If I can be of any help, let me know.
Thank you 
Jones

Craft: `v3.4.x-dev`
Php: `7.2`
Plugins:
```
    "craftcms/element-api": "2.5.4",
    "craftcms/redactor": "2.3.3.2",
    "mmikkel/cp-field-inspect": "2.0.0-beta.6",
    "nystudio107/craft-seomatic": "3.2.28",
    "putyourlightson/craft-cp-style-guide": "1.0.2",
    "sebastianlenz/linkfield": "1.0.19",
    "spicyweb/craft-neo": "^2.5.5",
    "verbb/field-manager": "2.1.0",
    "vlucas/phpdotenv": "^2.4.0"
```

**Comment by mmikkel** [comment]
Hi @Jones-S,

Unable to reproduce this on a fresh 3.4.0-alpha.1 install with the following plugins enabled:

![image](https://user-images.githubusercontent.com/298510/67211089-77569200-f41a-11e9-9a9c-51c6c1d52606.png)

Regarding https://github.com/craftcms/cms/issues/4944; CP Field Inspect doesn't actually interact with the Fields service until after all plugins have loaded (i.e. in an `EVENT_AFTER_PLUGINS_LOADED` event handler), so it's weird if it's in fact the reason for this error.

Do you have any additional plugins installed, beyond what you've listed above? Do your installed plugins' versions correspond to mine?

Also, does the error go away if you disable or uninstall CP Field Inspect (you can add the below to your `config/general.php` file to programmatically disable the plugin)?

```
<?php
    '*' => [
        'disabledPlugins' => ['cp-field-inspect'],
    ],
```

**Comment by mmikkel** [comment]
@Jones-S Update: Did some more digging and testing and was eventually able to reproduce the issue (it seems that the order in which CP Field Inspect and Typed Link Field is installed makes a difference – I was only able to reproduce it when Typed Link Field appears below CP Field Inspect in the `craft_plugins` table.

Anyway, just published [CP Field Inspect 1.0.7](https://github.com/mmikkel/CpFieldInspect-Craft/releases/tag/1.0.7), which should resolve the issue. Thanks for reporting!

**Comment by Jones-S** [comment]
Thanks for the quick fix. 👍
